### PR TITLE
Added Rejected workflow fixes #3003

### LINKED ISSUE #3003: Rapid detail page state for rejected experiments
If a reviewer rejects an experiment in remote settings, a rejection reason will be entered in the remote settings admin.  We can then mark the experiment in Experimenter as rejected and show the rejection reason, and in theory the owner should be able to modify their experiment accordingly and then resubmit for review.  We'll need workflow and UX/UI for this.

### PR DESCRIPTION
**Pull request by jaredlockhart**
Taking over for @tiftran from https://github.com/mozilla/experimenter/pull/3154

### CODE REVIEW

File: app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx [hunk 1]
```diff
@@ -131,34 +131,35 @@ const ExperimentDetails: React.FC = () => {
...
     changeStatusButtons = (
       <div className="d-flex mt-4">
         <span>
-          <Link
-            className={buttonsClass}
-            to={buttonsDisabled ? "#" : `/${experimentData.slug}/edit/`}
-          >
-            Back
+          <Link to={`/${experimentData.slug}/edit/`}>
+            <button className="btn btn-primary" disabled={backButtonDisabled}>
```
Comment (k88hudson) [inline app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx:153]: Why did you change this to a button?

Comment (jaredlockhart) [reply app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx:153]: Oh I didn't notice she did that, that's clever!  This is so it gets properly disabled rather than 'fake' disabled which was driving @pdehaan nuts

File: app/experimenter/static/rapid/components/forms/ExperimentForm.tsx [hunk 2]
```diff
@@ -37,7 +38,10 @@ const ErrorList: React.FC<ErrorListProperties> = ({ errors }) => {
 };
 
 const ExperimentForm: React.FC = () => {
-  const formData = useExperimentState();
+  // eslint-disable-next-line
```
Comment (k88hudson) [inline app/experimenter/static/rapid/components/forms/ExperimentForm.tsx:41]: What are you disabling here?

Comment (jaredlockhart) [reply app/experimenter/static/rapid/components/forms/ExperimentForm.tsx:41]: Oh interesting, i'll check that out

File: app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx (hunk 1 above)
Comment (jaredlockhart) [reply app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx:153]: Oh it doesn't actually solve the problem, it still shows the hand cursor.

Comment (jaredlockhart) [reply app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx:153]: Sigh we probably have to write some css for that.

Comment (k88hudson) [reply app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx:153]: i'll take a look when i run it

Comment (k88hudson) [reply app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx:153]: Ok, this is what I'd do:

```ts
<Link
  className={
    "btn btn-primary" + (backButtonDisabled ? " disabled" : "")
  }
  to={`/${experimentData.slug}/edit/`}
  onClick={(e) => backButtonDisabled && e.preventDefault()}
  >
  Back
</Link>
```

File: app/experimenter/static/rapid/components/forms/ExperimentForm.tsx (hunk 2 above)
Comment (k88hudson) [reply app/experimenter/static/rapid/components/forms/ExperimentForm.tsx:41]: Ah, I see what's going on. I'd recommend changing this to

```ts
const formData = {...useExperimentState(), status: ExperimentStatus.DRAFT}
```

Comment (pdehaan) [reply app/experimenter/static/rapid/components/forms/ExperimentForm.tsx:41]: +1. Also, it's usually more helpful to just ignore/disable a specific ESLint rule on the next line, if possible (vs shotgun disabling everything). But of course now that I've said that, I'll have to do some <kbd>git grep</kbd> to check our current eslint disable usages or the curiosity will drive me crazy.

File: app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx [hunk 3]
```diff
@@ -169,6 +170,21 @@ const ExperimentDetails: React.FC = () => {
...
 
+  let rejectFeedback;
+
+  if (experimentData.reject_feedback) {
+    const messageDate = new Date(experimentData.reject_feedback.changed_on);
+    rejectFeedback = (
+      <>
+        <h3 className="my-4">Review Feedback</h3>
+        <div className="alert alert-secondary" role="alert">
+          <p className="font-weight-bold"> {messageDate.toDateString()}</p>
```
Comment (pdehaan) [inline app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx:181]: Why the space before the date string?

File: app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx [hunk 4]
```diff
@@ -169,6 +170,21 @@ const ExperimentDetails: React.FC = () => {
...
+          <p>Reject reason: {experimentData.reject_feedback.message}</p>
```
Comment (pdehaan) [inline app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx:182]: Is the expectation that we'd always have a single line of feedback from Remote Settings?
Also, a bit extreme, but do we need to make sure this is sanitized at all since it's coming from Remote Settings and we aren't going to XSS ourselves or something? (Although I guess if somebody already has access to our Remote Settings bucket, having them return a reject reason with potential `<script>` tags is probably the least of our worries)

File: app/experimenter/static/rapid/components/forms/ExperimentForm.tsx (hunk 2 above)
Comment (jaredlockhart) [reply app/experimenter/static/rapid/components/forms/ExperimentForm.tsx:41]: @k88hudson that doesn't work because then you're including `state` twice in the object and TS complains, once in the first and then explicitly in the second.

Comment (jaredlockhart) [reply app/experimenter/static/rapid/components/forms/ExperimentForm.tsx:41]: I can't see an easy way to do this so I'm just gonna leave it how @tiftran wrote it.

File: app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx (hunk 4 above)
Comment (jaredlockhart) [reply app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx:182]: Both fine points but seems to support multiple lines and properly escape characters:

Screenshot_2020-07-30 Mozilla Experimenter(6)

File: app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx (hunk 3 above)
Comment (jaredlockhart) [reply app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx:181]: 🤷‍♀️

File: app/experimenter/kinto/tasks.py [hunk 5]
```diff
@@ -25,13 +29,14 @@ def push_experiment_to_kinto(experiment_id):
...
 
     experiment = Experiment.objects.get(id=experiment_id)
-    ExperimentBucketNamespace.request_namespace_buckets(
-        experiment.normandy_slug,
-        experiment,
-        NIMBUS_DATA["ExperimentDesignPresets"]["empty_aa"]["preset"]["arguments"][
-            "bucketConfig"
-        ]["count"],
-    )
+    if not ExperimentBucketRange.objects.filter(experiment=experiment).exists():
```
[the remark this answers is not in the document]
Comment (jaredlockhart) [reply app/experimenter/kinto/tasks.py:32]: Ya

File: app/experimenter/static/rapid/components/forms/ExperimentForm.tsx (hunk 2 above)
Comment (jaredlockhart) [reply app/experimenter/static/rapid/components/forms/ExperimentForm.tsx:41]: Oh it's htis:

```js
  const formData = {
    ...useExperimentState(),
    ...{ status: ExperimentStatus.DRAFT },
  };
```

Comment (k88hudson) [reply app/experimenter/static/rapid/components/forms/ExperimentForm.tsx]: Really? `const formData = {...useExperimentState(), status: ExperimentStatus.DRAFT}` works for me (they should be equivalent?)
